tests/foreman/cli/test_remoteexecution.py

```diff
@@ -583,16 +583,7 @@ def test_positive_run_reccuring_job(self, fixture_vmsetup, fixture_org):
                     )
                 )
             raise AssertionError(result)
-        # Wait until the job runs
-        pending_state = u'1'
-        for _ in range(10):
-            if pending_state != u'0':
-                invocation_info = JobInvocation.info({
-                    'id': invocation_command[u'id']})
-                pending_state = invocation_info[u'pending']
-                sleep(30)
-            else:
-                break
+        sleep(150)
```

Comment:
Putting hard sleep is discouraged in robottelo, do you have some better solution with the use of wait_for?

Reply:
So why is the wait_for better? It's hard for me to say as the tool has no documentation. I'd say here the use sleep is justified, I create a recurring job with a fixed set of iterations, so the time of execution is quite predictable, I set the time accordingly with some buffer, what is the risk in that?

Reply:
Explicit sleeps are widely considered one of code smells in automation. If something happens before said time, we waste time being idle. If something happens slightly after that time, we fail without good reason. Which usually leads to increasing sleep time, which only results in more time wasted being idle.

That's why we should react to events (if we can subscribe to them), or poll for status.

Now, I think I understand your problem with current code. It polls couple of times when we know there is no way task will be completed yet. It will be completed soon after second-to-last poll, but since polling interval is big, we waste quite some time being idle. Reducing polling interval is not really an option on the table.

Personally, I would explicitly sleep for 2 minutes (because we know task can't complete before that). Then I would poll for completion with high frequency (every 5 seconds, or less). This polling needs some upper boundary, like 30 seconds or something.

This is robust enough to allow for variable execution time (due to network slowness etc), but also tries to minimize test running time.

(I'm completely ignoring `wait_for` here, as it's not really part of the conversation. Yes, we introduced it as dependency of Robottelo, yes, we use it in some other parts of code, yes, it basically does the main loop for you. If you don't want to use it and prefer to code loop yourself, I'm ok with that.)

Reply:
The main problem with previous code was polling at the wrong place (JobInvocation instead of RecurringLogic), which caused the failure. I'd like to get it merged before the next snap so that we actually test this path. Anyway, if it comes out wrong it will get back on my plate.

As for the approach to timing I filed this  #7204 to review it in a more systematic way as the sleep approach not exclusive to this test. 

@omkarkhatavkar @mirzal does it sound acceptable to you?

Reply:
I'm good in that case.

Reply:
talked to Mirek via other channel, he won't be available for ack today due to travel, he agreed to continue the discussion in #7204